### PR TITLE
Support more types in MurmurhashOp

### PR DESCRIPTION
**Pull request by wangyems**
**Description**: Support Int64/UInt64, Float, Double in MurmurHash Op

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/murmur_hash3.cc [hunk 1]
```diff
@@ -193,14 +197,14 @@ Status MurmurHash3::Compute(OpKernelContext* ctx) const {
       ++output;
     }
   } else {
-    auto input = reinterpret_cast<const uint32_t*>(keys->DataRaw());
-    const auto input_end = input + input_count;
+    auto input = reinterpret_cast<const unsigned char*>(keys->DataRaw());
+    const auto input_end = input + input_count * static_cast<int>(input_element_bytes);
```
Comment (wschin) [inline onnxruntime/contrib_ops/cpu/murmur_hash3.cc:201]: ORT_ENFORCE(input_element_bytes < INT_MAX)

File: onnxruntime/contrib_ops/cpu/murmur_hash3.cc [hunk 2]
```diff
@@ -193,14 +197,14 @@ Status MurmurHash3::Compute(OpKernelContext* ctx) const {
       ++output;
     }
   } else {
-    auto input = reinterpret_cast<const uint32_t*>(keys->DataRaw());
-    const auto input_end = input + input_count;
+    auto input = reinterpret_cast<const unsigned char*>(keys->DataRaw());
```
Comment (wschin) [inline onnxruntime/contrib_ops/cpu/murmur_hash3.cc:200]: Why `unsigned char`? I remember Murmur hashing is based on `uint32`.

File: onnxruntime/contrib_ops/cpu/murmur_hash3.cc [hunk 3]
```diff
@@ -193,14 +197,14 @@ Status MurmurHash3::Compute(OpKernelContext* ctx) const {
...
+    const auto input_end = input + input_count * static_cast<int>(input_element_bytes);
     while (input != input_end) {
       MurmurHash3_x86_32(input,
                          static_cast<int>(input_element_bytes),
                          seed_,
                          output);
-      ++input;
+      input += static_cast<int>(input_element_bytes);
```
Comment (wschin) [inline onnxruntime/contrib_ops/cpu/murmur_hash3.cc:207]: Convert once `static_cast<int>(input_element_bytes)` in the beginning and reuse its value.

File: onnxruntime/test/contrib_ops/murmur_hash3_test.cc [hunk 4]
```diff
@@ -49,14 +49,46 @@ TEST(MurmurHash3OpTest, ZeroSeedUIntResult2) {
...
+  test.AddAttribute<int64_t>("seed", 0LL);
+  test.AddOutput<uint32_t>("Y", {1}, {6814352L});
+  test.Run();
+}
+
+TEST(MurmurHash3OpTest, ZeroSeedDoubleResult) {
+  OpTester test("MurmurHash3", 1, onnxruntime::kMSDomain);
+  test.AddInput<double>("X", {1}, {3.});
+  test.AddAttribute<int64_t>("seed", 0LL);
+  test.AddOutput<uint32_t>("Y", {1}, {3554953595L});
```
Comment (wschin) [inline onnxruntime/test/contrib_ops/murmur_hash3_test.cc:72]: Where does it come from? Scikit-learn? Pytorch? Tensorflow?

Comment (wangyems) [reply onnxruntime/test/contrib_ops/murmur_hash3_test.cc:72]: it's for supporting ML.NET(verified compatibility) and there is no source for test cases.based on offline chat

File: onnxruntime/contrib_ops/cpu/murmur_hash3.cc (hunk 2 above)
Comment (wangyems) [reply onnxruntime/contrib_ops/cpu/murmur_hash3.cc:200]: cast to uint32 when calling MurmurHash3_x86_32

File: onnxruntime/contrib_ops/cpu/murmur_hash3.cc [hunk 5]
```diff
@@ -193,14 +197,16 @@ Status MurmurHash3::Compute(OpKernelContext* ctx) const {
...
-    auto input = reinterpret_cast<const uint32_t*>(keys->DataRaw());
-    const auto input_end = input + input_count;
+    auto input = reinterpret_cast<const unsigned char*>(keys->DataRaw());
+    ORT_ENFORCE(input_element_bytes < static_cast<size_t>(std::numeric_limits<int>::max()));
+    int input_num_bytes = static_cast<int>(input_element_bytes);
+    const auto input_end = input + input_count * input_num_bytes;
     while (input != input_end) {
-      MurmurHash3_x86_32(input,
-                         static_cast<int>(input_element_bytes),
+      MurmurHash3_x86_32(reinterpret_cast<const uint32_t*>(input),
```
Comment (wschin) [inline onnxruntime/contrib_ops/cpu/murmur_hash3.cc:205]: What if `input_num_bytes < sizeof(int)`? Would it access some random bits?

Comment (wangyems) [reply onnxruntime/contrib_ops/cpu/murmur_hash3.cc:205]: I think that e.g if we reinterpret_cast a uint8 to uint32, the uint8 will be padded into a uint32 without access random bits

File: onnxruntime/contrib_ops/cpu/murmur_hash3.cc [hunk 6]
```diff
@@ -193,14 +197,16 @@ Status MurmurHash3::Compute(OpKernelContext* ctx) const {
       ++output;
     }
   } else {
-    auto input = reinterpret_cast<const uint32_t*>(keys->DataRaw());
-    const auto input_end = input + input_count;
+    auto input = reinterpret_cast<const unsigned char*>(keys->DataRaw());
+    ORT_ENFORCE(input_element_bytes < static_cast<size_t>(std::numeric_limits<int>::max()));
```
Comment (wschin) [inline onnxruntime/contrib_ops/cpu/murmur_hash3.cc:201]:
```suggestion
    ORT_ENFORCE(input_element_bytes % 4 == 0);
```

**Pull request merged**